### ISSUE TITLE
Slow download speed from internet and cache

### ISSUE THREAD

**Issue by nomadics9** [lancachenet/lancache-dns#105]
Good evening guys,

I am setting up lancache server for my internet cafe with basic linux knowledge I have installed lancache monolithic and even tried generic for steam,blizzard,riot,uplay,epic and origin.

With everything working fine and as intended tested with blizzard and steam but I have been facing some problems.

The download speed if the game isnt cached through blizzard launcher is slow also steam but not as much as 1mb/s with battlenet launcher. 

I have been searching and watching every tutorial to try and fix my problem before coming here. I ended up with using 36 ips 6 ips for each. With no success tested with generic and monolithic what I noticed is it looked like there is something limiting the cache speed to 10MB/s if I started to download from cache with 2 clients its split to 5 5. And monitoring with nload.

As of right now im testing the server to see how it works and that is working as intended before I go and buy a dedicated server im using one of the client pcs for that with 250gb ssd and 16gb ram i5 9600k. Might be hardware issue but my first try with monolithic i did successfully made it work with 50mb/s but after fresh install again to test with other launchers im facing this problem.

I have no clue on how to configure anything at all besides installing the dockers with some configuration in the run command.

My server right now is running:
Monolithic with 36 ips 
Lancache-dns with 6 ips for each launcher not using generic cache
And sniproxy which I dont fully understand but I think it just to redirect https links. 

Tldr;

Everything working as intended but download speed for uncached games is slow and download speed for cached games is getting capped to 10MB/s. Even though one of the tests I did to successfully get 50MB/s. 

Im using 1GB lan network.

I would really appreciate any help.

I would have posted all my configuration with the dockers here but I cant access them right now and this is all what im thinking about.

Thank you!

**Comment by nomadics9** [comment]
this is the notepad where I arrange the commands before I run them
+++++++STEAM+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name steam-cache \
  -v /home/sager/cache/steam/data:/data/cache \
  -v /home/sager/cache/steam/logs:/data/logs \
  -p 192.168.100.113:80:80 -p 192.168.100.114:80:80 -p 192.168.100.115:80:80 -p 192.168.100.116:80:80 -p 192.168.100.117:80:80 -p 192.168.100.118:80:80 -p 192.168.100.119:80:80 \
  lancachenet/generic:latest

+++++++++BLIZZARD++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name blizzard-cache \
  -v /home/sager/cache/blizzard/data:/data/cache \
  -v /home/sager/cache/blizzard/logs:/data/logs \
  -p 192.168.100.120:80:80 -p 192.168.100.121:80:80 -p 192.168.100.122:80:80 -p 192.168.100.123:80:80 -p 192.168.100.124:80:80 -p 192.168.100.125:80:80 -p 192.168.100.126:80:80 \
  lancachenet/generic:latest

++++++++RIOT+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name riot-cache \
  -v /home/sager/cache/riot/data:/data/cache \
  -v /home/sager/cache/riot/logs:/data/logs \
  -p 192.168.100.127:80:80 -p 192.168.100.128:80:80 -p 192.168.100.129:80:80 -p 192.168.100.130:80:80 -p 192.168.100.131:80:80 -p 192.168.100.132:80:80 -p 192.168.100.133:80:80 \
  lancachenet/generic:latest

++++++++++UPLAY++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name uplay-cache \
  -v /home/sager/cache/uplay/data:/data/cache \
  -v /home/sager/cache/uplay/logs:/data/logs \
  -p 192.168.100.134:80:80 -p 192.168.100.135:80:80 -p 192.168.100.136:80:80 -p 192.168.100.137:80:80 -p 192.168.100.138:80:80 -p 192.168.100.139:80:80 -p 192.168.100.140:80:80  \
  lancachenet/generic:latest

+++++++++++EPIC++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name epic-cache \
  -v /home/sager/cache/epic/data:/data/cache \
  -v /home/sager/cache/epic/logs:/data/logs \
  -p 192.168.100.141:80:80 -p 192.168.100.142:80:80 -p 192.168.100.143:80:80 -p 192.168.100.144:80:80 -p 192.168.100.145:80:80 -p 192.168.100.146:80:80 -p 192.168.100.147:80:80  \
  lancachenet/generic:latest

++++++++++ORIGIN++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name origin-cache \
  -v /home/sager/cache/origin/data:/data/cache \
  -v /home/sager/cache/origin/logs:/data/logs \
  -p 192.168.100.148:80:80 -p 192.168.100.149:80:80 -p 192.168.100.150:80:80 -p 192.168.100.151:80:80 -p 192.168.100.152:80:80 -p 192.168.100.153:80:80 \
  lancachenet/generic:latest

++++++++++LANCACHE-DNS+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
  --restart unless-stopped -d \
  --name lancache-dns \
  -p 192.168.100.113:53:53/udp \
  -e UPSTREAM_DNS=1.1.1.1 \
  -e UPSTREAM_DNS2=1.0.0.1 \
  -e STEAMCACHE_IP="192.168.100.113 192.168.100.114 192.168.100.115 192.168.100.116 192.168.100.117 192.168.100.118 192.168.100.119" \
  -e ORIGINCACHE_IP="192.168.100.148 192.168.100.149 192.168.100.150 192.168.100.151 192.168.100.152 192.168.100.153" \
  -e BLIZZARDCACHE_IP="192.168.100.120 192.168.100.121 192.168.100.122 192.168.100.123 192.168.100.124 192.168.100.125 192.168.100.126" \
  -e RIOTCACHE_IP="192.168.100.127 192.168.100.128 192.168.100.129 192.168.100.130 192.168.100.131 192.168.100.132 192.168.100.133" \
  -e EPICGAMESCACHE_IP="192.168.100.141 192.168.100.142 192.168.100.143 192.168.100.144 192.168.100.145 192.168.100.146 192.168.100.147" \
  -e UPLAYCACHE_IP="192.168.100.134 192.168.100.135 192.168.100.136 192.168.100.137 192.168.100.138 192.168.100.139 192.168.100.140" \
  lancachenet/lancache-dns:latest

+++++++++SNIPROXY+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

sudo docker run \
 --restart unless-stopped -d \
  --name sniproxy \
  -p 192.168.100.113:443:443 \
  lancachenet/sniproxy:latest

**Comment by nomadics9** [comment]
Issue been resolved, Only getting slow uncache speed with battlenet right now everything else working fine.

if anyone is interested I changed where sniproxy is pointing to -p :443:443 instead of putting the ip there.

**Comment by unspec** [comment]
Regarding slow initial blizzard downloads:

The latest version of monolithic/generic now supports changing the slice size used by nginx. We've found that increasing from 1m to 8m offers a small performance boost to specific use cases (single user initial downloads of blizzard games in particular). See http://lancache.net/docs/advanced/tuning-cache/#tweaking-slice-size for information on how to make use of this.

Please note that it does come with some potential downsides (discussed in the above link) and will invalidate any already cached data on your cache if you change the value.

To tidy up the issues, if you choose to test this please post any feedback on this issue: https://github.com/lancachenet/generic/issues/100

If you need any other support please see http://lancache.net/support/ or open a new issue.
